### PR TITLE
feat: add logging for testing background queue

### PR DESCRIPTION
**Pull request by levibostian**
Improved the logging of the background queue and parts of the SDK to make QA testing better. 

[How to enable logging to test this PR](https://github.com/customerio/RemoteHabits-iOS/pull/33)

### CODE REVIEW

File: Sources/MessagingPush/MessagingPushImplementation.swift [hunk 1]
```diff
@@ -49,12 +57,17 @@ internal class MessagingPushImplementation: MessagingPushInstance {
...
     public func deleteDeviceToken() {
+        logger.info("deleting device token request made")
+
         guard let existingDeviceToken = globalDataStore.pushDeviceToken else {
+            logger.info("no device token exists so ignoring request to delete")
             return // no device token to delete, ignore request
         }
-        globalDataStore.pushDeviceToken = nil
+        // Do not delete push token from device storage. The token is valid
+        // once given to SDK. We need it for future profile identifications.
```
Comment (levibostian) [inline Sources/MessagingPush/MessagingPushImplementation.swift:67]: this is a bug that I found after implementing logging and I did some QA testing of the queue myself.

File: Sources/MessagingPush/MessagingPushImplementation.swift [hunk 2]
```diff
@@ -24,18 +27,23 @@ internal class MessagingPushImplementation: MessagingPushInstance {
...
     public func registerDeviceToken(_ deviceToken: String) {
+        logger.info("registering device token \(deviceToken)")
+
+        logger.debug("storing device token to device storage \(deviceToken)")
         // no matter what, save the device token for use later. if a customer is identified later,
         // we can reference the token and register it to a new profile.
         globalDataStore.pushDeviceToken = deviceToken
 
         guard let identifier = profileStore.identifier else {
+            logger.info("no profile identified so not registering device token a profile")
```
Comment (hownowstephen) [inline Sources/MessagingPush/MessagingPushImplementation.swift:46]:
```suggestion
            logger.info("no profile identified, so not registering device token to a profile")
```

File: Sources/Tracking/CustomerIOImplementation.swift [hunk 3]
```diff
@@ -91,9 +99,11 @@ public class CustomerIOImplementation: CustomerIOInstance {
         // don't modify the state of the SDK until we confirm we added a background queue task successfully.
         // XXX: better handle scenario when adding task to queue is not successful
         if queueStatus.success {
```
Comment (hownowstephen) [inline Sources/Tracking/CustomerIOImplementation.swift:101]:
```suggestion
        if !queueStatus.success {
            // XXX: better handle scenario when adding task to queue is not successful
            logger.debug("failed to enqueue identify task")
            return
        }
```

Comment (hownowstephen) [reply Sources/Tracking/CustomerIOImplementation.swift:101]: would reduce nesting of the happy path below, and make the comment a little clearer

File: Sources/Tracking/CustomerIOImplementation.swift [hunk 4]
```diff
@@ -128,6 +145,7 @@ public class CustomerIOImplementation: CustomerIOInstance {
             // XXX: log error for customer to debug their request body
```
Comment (hownowstephen) [inline Sources/Tracking/CustomerIOImplementation.swift:145]: log here

File: Sources/Tracking/CustomerIOImplementation.swift [hunk 5]
```diff
@@ -117,6 +132,8 @@ public class CustomerIOImplementation: CustomerIOInstance {
         data: RequestBody,
         jsonEncoder: JSONEncoder? = nil
     ) {
+        logger.info("tracking event \(name)")
+
         guard let currentlyIdentifiedProfileIdentifier = profileStore.identifier else {
             // XXX: when we have anonymous profiles in SDK,
             // we can decide to not ignore events when a profile is not logged yet.
```
Comment (hownowstephen) [inline Sources/Tracking/CustomerIOImplementation.swift:139]: debug log here?

File: Sources/Tracking/Util/Log.swift [hunk 6]
```diff
@@ -1,34 +1,96 @@
...
-        print("\(prefix) Info: \(message)")
+    #if canImport(os)
+    // Unified logging for Swift. https://www.avanderlee.com/workflow/oslog-unified-logging/
+    // This means we can view logs in xcode console + Console app.
+    private func printMessage(_ message: String, _ level: OSLogType) {
+        if !minLogLevel.shouldLog(level) { return }
+
+        if #available(iOS 14, *) {
+            let logger = os.Logger(subsystem: self.logSubsystem, category: self.logCategory)
+            logger.info("\(message, privacy: .public)")
```
Comment (hownowstephen) [inline Sources/Tracking/Util/Log.swift:70]: how come we're not using the os.Logger / OSLog levels here, and doing everything at .info? `level` seems unused?

**Review by hownowstephen — APPROVED** [review]

**Pull request merged**